cpp/src/arrow/testing/gtest_util.h

```diff
@@ -228,7 +253,7 @@ void ChunkedArrayFromVector(const std::shared_ptr<DataType>& type,
                             const std::vector<std::vector<C_TYPE>>& values,
                             std::shared_ptr<ChunkedArray>* out) {
   ArrayVector chunks;
-  DCHECK_EQ(is_valid.size(), values.size());
+  ASSERT_EQ(is_valid.size(), values.size());
```

Comment:
do these always work as long as they are in a void function or do they need to be called directly from a test?

Reply:
Hmm... good question. They probably need to be called from a test.

Reply:
So I think I'll tackle ARROW-5145 first.

Reply:
OTOH, the called code also invokes `ASSERT_OK`, so we should be good.